README.md

```diff
@@ -1,89 +1,218 @@
 # Summa Aggregation
 
+Summa Aggregation is focused on optimizing the generation of Merkle sum tree, a task identified as the primary time-consuming process in Summa benchmarks. Our goal is to significantly reduce the time required to generate these tree by leveraging parallelization across multiple machines.
+
+The system features `AggregationMerkleSumTree`, a specialized component designed for the efficient construction of complete tree from smaller, aggregated structures known as `mini-tree`. This approach not only speeds up the tree-building process but also enhances scalability and performance in large-scale data environments.
+
 ## Orchestrator
 
-WIP
+The Orchestrator in the Summa Aggregation serves as the central management component, coordinating the data processing activities. It plays a pivotal role in coordinating the activities of `Executors` and `Workers`, improving of tasks in the generation of Merkle sum tree.
+Key functions of the `Orchestrator` include:
+
+- **Dynamic Executor Spawning**: The Orchestrator dynamically spawns `Executors` in numbers set by the user. Each `Executor` is then connected to a dedicated `Worker` for efficient task execution.
+
+- **Task Management and Distribution**: It oversees the overall task flow, loading tasks and distributing them to `Executors`.
+
+- **Error Management and Pipeline Control**: The Orchestrator handles basic pipeline control and responds to errors by initiating the cancellation of all tasks.
+
+## Executor and Worker
+
+The `Executor` acts as a crucial intermediary between the `Orchestrator` and `Workers`, facilitating the data processing workflow. Spawned by the `Orchestrator`, each `Executor` operates in a one-to-one relationship with a `Worker`. The `Worker` is a server that can generate a merkle sum tree, internally called a `mini-tree`, by receiving entries data. The primary role of a `Worker` is to build these `mini-trees`, which are segments of the `AggregationMerkleSumTree`.
+
+Key aspects of the `Executor's` role include:
+
+- **Spawning and Connection**: `Executors` are dynamically spawned by the `Orchestrator` as part of the system's scalability. Each `Executor` is designed to connect with a `Worker` for task execution.
+
+- **Data Handling and Task Distribution**: A primary function of the `Executor` is to receive data entries, often parsed and prepared by the Orchestrator. Upon receiving these entries, the Executor is responsible for forwarding them to its connected `Worker`.
+
+- **Communication Bridge**: The `Executor` serves as a communication bridge within the data pipeline. It relays processed data, `mini-trees`, from Workers back to the Orchestrator.
+
+The `Worker` called in here, mostly point out the container that runs `mini-tree-generator` server.
+
```

Comment:
Sorry for the misuse of the definite article. 
The Worker is actually a Docker container, its image can be built from `Dockerfile`.
And the container runs `mini-tree-server` internally.
I will fix `mini-tree-server` instead of `mini-tree-generator`.